### DISCUSSION TITLE
Proposal: module metadata

### DISCUSSION THREAD

**Discussion by rogpeppe** [cue-lang/cue#3057] [Proposal]
This discussion related to the overarching module v3 proposal (https://github.com/cue-lang/cue/discussions/2939) and is about the proposal as to how to include metadata about a CUE module. The full proposal is at: https://github.com/cue-lang/proposal/blob/main/designs/modules.v3/3057-module-metadata.md

**Comment by myitcv** [comment]
On the [2024-05-16 Modules update call](https://github.com/cue-lang/cue/discussions/3119) we [accepted](https://docs.google.com/presentation/d/17VtQs3QitwXFTBI5BT69h6MzPvEEOlqJtQ7XYCExl7M/edit#slide=id.g2bbab775c99_0_34) [v3 of the modules proposal and its sub-proposals](https://github.com/cue-lang/cue/discussions/2939). Closing this discussion to recognise that decision.

**Comment by infogulch** [comment]
Have we considered adding a module format version somewhere? It could allow for changing the module format in the future while remaining backwards compatible in case something comes up in the future.

**Comment by rogpeppe** [comment]
@infogulch
> Have we considered adding a module format version somewhere? It could allow for changing the module format in the future while remaining backwards compatible in case something comes up in the future.

The module manifest includes a media type (`application/vnd.cue.module.v1+json`), part of which is a version. I believe that's sufficient, but if you foresee any issues with that, please say so.
